src/setting_data.c

```diff
@@ -64,8 +64,8 @@ setting_data_t *setting_data_create(type_data_t *type_list,
     free(setting_data);
     setting_data = NULL;
   } else {
-    strncpy(setting_data->section, section, section_len);
-    strncpy(setting_data->name, name, name_len);
+    strcpy(setting_data->section, section);
```

Comment:
we should probably avoid `strcpy` (since certain customers explicitly ban `strcpy` from being referenced in a binary), would `strncpy(setting_data->section, section, section_len-1)` also work to fix the error?

Reply:
I thought you would say this.

`strncpy(setting_data->section, section, section_len-1)` would leave the possibility of the string not being NULL-terminated.

It can be fixed with an explicit NULL assignment on the following line, but it's unnecessarily complicated/less efficient than just using `strcpy()`.

Reply:
😄 -- it's not a hot path function, so I would prefer the strncpy + manual termination